src/core/qgsfield.cpp

```diff
@@ -268,6 +268,11 @@ QString QgsField::displayString( const QVariant &v ) const
   {
     return QObject::tr( "BLOB" );
   }
+  else if ( d->type == QVariant::StringList )
+  {
+    return "{" + v.toStringList().join( "," ) + "}";
```

Comment:
```suggestion
    return '{' + v.toStringList().join( ',' ) + '}';
```

Reply:
What do you think about json format `[x,y,z]`?

Reply:
> But if the data comes from JSON (is it possible?), it will be the opposite problem.

Yes https://qgis.org/en/site/forusers/visualchangelog34/#feature-json-jsonb-type-support

Reply:
Since this change is not inside the postgres provider, I am not sure we should take it as reference.

Reply:
I understand your point, but I can't find an elegant way of fixing this.

Best I can propose is to add a method on QgsField to setup the way we have to format array. To set the default to ``[`` and ``]`` and call this method from postgres provider.

Reply:
Is the underlying problem that the postgres provider is only able to parse non-nested arrays?

Reply:
Sorry for the silence, let's see if we can define some basic requirements

 * internally in QGIS the data should be passed around as QVariantList (of QVariantLists) to allow for generic implementation for data aggregation and the like
 * the "default" representation should be as generic as it gets (and my gut feeling is that json is more in this area than hstore formatting) but having this configurable (or autoconfigured) sounds good

... so re-reading your proposal https://github.com/qgis/QGIS/pull/9610#discussion_r273470112 sounds quite good, I must have misread that before, sorry.

Reply:
I made the modifications described above 24fd6ce
> Best I can propose is to add a method on QgsField to setup the way we have to format array. To set the default to [ and ] and call this method from postgres provider.